### PR TITLE
Add a skeleton JS project that allows all ESM syntax

### PR DESCRIPTION
**Pull request by jamie-lynch**
## What does this change?

Many candidates may be familiar with using ES modules, rather than common JS (i.e. `import something from "somewhere"` over `const something = require("somewhere")`). 

This PR adds a new `javascript-es6` directory which contains a JS skeleton project that allows the use of all ES6 features. 

Similar to the other directories, this project provides:

* An `index.js` file for source code to live in, with a basic `pairingTest` function
* An `index.test.js` file for tests to be written in, with a failing test
* Scripts to setup and test the project

I've based this on the existing `javascript` project, with the following changes:

* Added `"type": "module"` to the `package.json`
* Installed`@babel/preset-env` and `babel-jest`
* Added a `.babelrc`
* Added some jest config to use babel to transform js files

## How to test

Pull down this branch and try running the setup, test and run scripts

## How can we measure success?

There is an environment where candidates can write JS and make use of ES modules syntax.

## Have we considered potential risks?

n/a

### CODE REVIEW

**Comment by mxdvl** [comment]
In this instance, does it make sense to keep the `yarn.lock` file? The interviewee might choose `npm` or `pnpm`, too.

**Comment by jamie-lynch** [comment]
> In this instance, does it make sense to keep the `yarn.lock` file? The interviewee might choose `npm` or `pnpm`, too.

Hmm... that's true. That said, this is currently consistent with the javascript and typescript projects so if we were to do that, I'd rather remove it from all three in a separate PR. 

We do also use `yarn` in the setup and test scripts, which gives slightly more reason to keep the lock file around as we're not completely agnostic about package managers at the moment.

**Comment by mxdvl** [comment]
Yeah I think if we want to keep lock files, we need to make sure we have a way to keep them up to date, maybe @dependabot.

**Review by mxdvl — APPROVED** [review]

File: javascript-esm/.nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v14.16.1
```
Comment (mxdvl) [inline javascript-esm/.nvmrc:1]: [Gallium is now in active LTS](https://nodejs.org/en/about/releases/)

```suggestion
v16.13.0
```

Comment (jamie-lynch) [reply javascript-esm/.nvmrc:1]: I've done this across all three JS/TS projects in https://github.com/guardian/coding-exercise-project/pull/73

**Pull request merged**